src/ocp_postprocess.rs

```diff
@@ -218,6 +236,156 @@ pub(crate) async fn fix_olm_secret_hash_annotation(in_memory_etcd_client: &Arc<I
+fn operator_log_level(log_level: &str) -> String {
+    match log_level {
+        "Normal" => "2",
+        "Debug" => "4",
+        "Trace" => "6",
+        "TraceAll" => "8",
+        _ => "2",
+    }
+    .to_string()
+}
+
+pub(crate) async fn fix_deployment_spec_hash_annotation(
+    in_memory_etcd_client: &Arc<InMemoryK8sEtcd>,
+    k8s_resource_location: K8sResourceLocation,
+) -> Result<()> {
+    let etcd_client = in_memory_etcd_client;
+
+    let mut deployment = get_etcd_json(etcd_client, &k8s_resource_location)
+        .await?
+        .context(format!("couldn't find {}", k8s_resource_location))?;
+    let dep = deployment.clone();
+
+    let revision = dep
+        .pointer("/metadata/labels/revision")
+        .context("no .metadata.labels.revision")?
+        .as_str()
+        .context("revision not a str")?;
+
+    let metadata_annotations = deployment
+        .pointer_mut("/metadata/annotations")
+        .context("no .metadata.annotations")?
+        .as_object_mut()
+        .context("annotations not an object")?;
+
+    let supported_namespaces = ["openshift-apiserver", "openshift-oauth-apiserver"];
+    assert!(
+        k8s_resource_location
+            .namespace
+            .as_ref()
+            .is_some_and(|ns| supported_namespaces.contains(&ns.as_str())),
+        "unsupported namespace {} for spec-hash annotation fix",
+        k8s_resource_location.namespace.unwrap()
+    );
+
+    match k8s_resource_location.namespace.as_deref() {
+        Some("openshift-apiserver") => {
+            let location = &K8sResourceLocation::new(
+                Some("openshift-apiserver-operator"),
+                "Deployment",
+                "openshift-apiserver-operator",
+                "apps/v1",
+            );
+            let deployment = get_etcd_json(etcd_client, location)
+                .await?
+                .context(format!("couldn't find {}", location))?;
+
+            let env = deployment
+                .pointer("/spec/template/spec/containers")
+                .context("no spec.template.spec.containers")?
+                .as_array()
+                .context("spec.template.spec.containers not an array")?
+                .first()
+                .context("spec.template.spec.containers is empty")?
+                .get("env")
+                .context("spec.template.spec.containers[0].env not found")?
+                .as_array()
+                .context("spec.template.spec.containers[0].env not an array")?;
+
+            let value = env.iter().find_map(|var| {
+                (var.as_object()?.get("name") == Some(&serde_json::Value::String("KUBE_APISERVER_OPERATOR_IMAGE".to_string())))
+                    .then_some(var.get("value")?)
+            });
+
+            let kube_apiserver_operator_image = value
+                .context("expected KUBE_APISERVER_OPERATOR_IMAGE to be in env vars")?
+                .as_str()
+                .context("value not a string")?;
```

Comment:
Maybe rust_analyzer extract function these (selected) lines into a function called `get_kube_apiserver_operator_image`?

Reply:
Absolutely, I'll update this.